rundetection/rules/loq_rules.py

```diff
@@ -0,0 +1,192 @@
+"""
+Rules for LOQ
+"""
+
+from __future__ import annotations
+
+import logging
+import typing
+from dataclasses import dataclass
+from pathlib import Path
+
+import requests
+import xmltodict
+
+from rundetection.rules.rule import Rule
+
+if typing.TYPE_CHECKING:
+    from rundetection.job_requests import JobRequest
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SansFileData:
+    title: str
+    type: str
+    run_number: str
+
+
+def _extract_run_number_from_filename(filename: str) -> str:
+    # Assume filename looks like so: LOQ00100002.nxs
+    # Return everything between 1 and 2 including 1 and 2.
+    filename = filename.split(".")[0]  # Drop .nxs
+    filename = filename.strip("LOQ")  # Drop LOQ
+
+    # Drop first 0s
+    run_number = ""
+    first_non_zero_found = False
+    for char in filename:
+        if first_non_zero_found is False and char != "0":
+            run_number += char
+            first_non_zero_found = True
+        elif first_non_zero_found:
+            run_number += char
+    return run_number
+
+
+def _extract_cycle_from_file_path(file_path: Path) -> str:
+    # Assume path looks like so: /archive/NDXLOQ/Instrument/data/cycle_24_2/LOQ000001.nxs
+    return file_path.parts[-2]
+
+
+def _is_sample_transmission_file(sans_file: SansFileData, sample_title: str) -> bool:
+    return sample_title in sans_file.title and sans_file.type == "TRANS"
+
+
+def _is_sample_direct_file(sans_file: SansFileData) -> bool:
+    return ("direct" in sans_file.title.lower() or "empty" in sans_file.title.lower()) and sans_file.type == "TRANS"
+
+
+def _is_can_scatter_file(sans_file: SansFileData, can_title: str) -> bool:
+    return can_title == sans_file.title.split("_")[0] and sans_file.type == "SANS/TRANS"
+
+
+def _is_can_transmission_file(sans_file: SansFileData, can_title: str) -> bool:
+    return can_title in sans_file.title and sans_file.type == "TRANS"
+
+
+def _find_trans_file(sans_files: list[SansFileData], sample_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_sample_transmission_file(sans_file=sans_file, sample_title=sample_title):
+            return sans_file
+    return None
+
+
+def _find_direct_file(sans_files: list[SansFileData]) -> SansFileData | None:
+    reversed_files = sans_files.copy()
+    reversed_files.reverse()
```

Comment:
More concise way of doing this?
```suggestion
    reversed_files = reversed(sans_files)
```

Reply:
Not only concise but more efficient.